command/agent/job_endpoint_test.go

```diff
@@ -1981,29 +1983,150 @@ func TestHTTP_JobDispatch(t *testing.T) {
 
 		// Make the HTTP request
 		req2, err := http.NewRequest(http.MethodPut, "/v1/job/"+job.ID+"/dispatch", buf)
+		must.NoError(t, err)
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Logf("err: %v", err)
 		}
 		respW.Flush()
 
 		// Make the request
 		obj, err := s.Server.JobSpecificRequest(respW, req2)
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Logf("err: %v", err)
 		}
 
 		// Check the response
 		dispatch := obj.(structs.JobDispatchResponse)
 		if dispatch.EvalID == "" {
-			t.Fatalf("bad: %v", dispatch)
+			t.Logf("bad: %v", dispatch)
 		}
-
+		must.NotEq(t, dispatch.EvalID, "")
 		if dispatch.DispatchedJobID == "" {
-			t.Fatalf("bad: %v", dispatch)
+			t.Logf("bad: %v", dispatch)
 		}
+		must.NotEq(t, dispatch.DispatchedJobID, "")
 	})
 }
 
+func TestHTTP_JobDispatchPriority(t *testing.T) {
+	ci.Parallel(t)
+	testCases := []struct {
+		name        string
+		priority    int
+		expectedErr bool
+		setPriority bool
+	}{{
+		name: "no priority",
+	},
+		{name: "set invalid priority",
+			priority:    103,
+			expectedErr: true,
+		},
+		{name: "set valid priority",
+			priority:    100,
+			setPriority: true,
+		},
+	}
```

Comment:
This test is great... but it doesn't test the new feature yet. I think we need a test case that covers setting the dispatch job priority to a different value than the parent job if we wanted to cover it.